### DISCUSSION TITLE
AGORA2 issues with .mat files

### DISCUSSION THREAD

**Discussion by katjadellalibera** [micom-dev/micom#184] [General]
Hello all,

I'm trying to use metabolic models from the AGORA 2 db (https://www.biorxiv.org/content/10.1101/2020.11.09.375451v1.full) .
When importing a .mat model file using load_model, I get a warning, but it does seem to work.

**[17:08:49] WARNING  No defined compartments in model model. Compartments will be deduced heuristically   
                    using regular expressions.**

If I then try to merge two models, I get an error.

**AttributeError: 'Model' object has no attribute 'decode'**

Any ideas how to fix this? Is it a problem with .mat files specifically? The xml files for AGORA2 unfortunately don't seem to download correctly and I've never had this issue with the original AGORA database.

Thank you in advance for your help.

- Katja

**Comment by cdiener** [comment]
Hi the warning is not a real issue because it will infer the compartments correctly for the AGORA models. Are you joining using `join_models`? This one expects a list of strings (paths to the model files). Maybe you were passing in model objects there?
